### PR TITLE
fix panic in cli arg parsing

### LINKED ISSUE #1735: gencmdref script ending in runtime error
I am trying to move `bin/gencmdref` to the ipfs/website directly, because the Makefile there uses it, it has a hardcoded to path to where it is stored on @jbenet's machine, and there's now been two PRs to the script in this repository to fix an issue in ipfs/website (#1733 and #1723). 

Running `ipfs commands | gencmdref` breaks, however, and I'm not sure why. Here is my exit code: 

```sh
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/ipfs/go-ipfs/commands/cli.parseArgs(0x8210fb4d0, 0x0, 0x3, 0x82111c008, 0xf41a80, 0x2, 0x2, 0x0, 0xf42220, 0x0, ...)
	/go/src/github.com/ipfs/go-ipfs/commands/cli/parse.go:266 +0x10ea
github.com/ipfs/go-ipfs/commands/cli.Parse(0x8210f23d0, 0x3, 0x3, 0x82111c008, 0xf42220, 0x0, 0x0, 0x821249d00, 0x0, 0x0, ...)
	/go/src/github.com/ipfs/go-ipfs/commands/cli/parse.go:45 +0x52b
main.(*cmdInvocation).Parse(0x8210f2840, 0x88212a2110, 0x8210fb440, 0x8210f23d0, 0x3, 0x3, 0x0, 0x0)
	/go/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:236 +0x89
main.main()
	/go/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:107 +0x3db

goroutine 18 [chan receive]:
github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/codahale/metrics.init.1.func2()
	/go/src/github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/codahale/metrics/metrics.go:321 +0x73
created by github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/codahale/metrics.init.1
	/go/src/github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/codahale/metrics/metrics.go:328 +0x7b

goroutine 19 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37
Traceback (most recent call last):
  File "gencmdref", line 38, in <module>
    main()
  File "gencmdref", line 28, in main
    print run((line + ' --help').split(' ')).strip()
  File "gencmdref", line 11, in run
    return check_output(cmd)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['ipfs', 'dht', 'put', '--help']' returned non-zero exit status 2
```

### PR DESCRIPTION
**Pull request by whyrusleeping**
fixes #1735 

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by whyrusleeping** [comment]
@noffle could I get you to CR this one?

File: core/commands/dht.go [hunk 1]
```diff
@@ -12,8 +12,8 @@ import (
 	notif "github.com/ipfs/go-ipfs/notifications"
 	path "github.com/ipfs/go-ipfs/path"
 	ipdht "github.com/ipfs/go-ipfs/routing/dht"
-	u "gx/ipfs/QmZNVWh8LLjAavuQ2JXuFmuYH3C11xo988vSgp7UQrTRj1/go-ipfs-util"
 	peer "gx/ipfs/QmUBogf4nUefBjmYjn6jfsfPJRkmDGSeMhNj4usRKq69f4/go-libp2p/p2p/peer"
+	u "gx/ipfs/QmZNVWh8LLjAavuQ2JXuFmuYH3C11xo988vSgp7UQrTRj1/go-ipfs-util"
```
Comment (noffle) [inline core/commands/dht.go]: Nitpick: I know gophers love short var names, but `u` -> `util` would be nice.

**Comment by noffle** [comment]
:horse: LGTM

This PR feels like a bandaid on top of some already fairly hard to follow code. It'd be nice if we could simplify the complex `if` structure, but this may be a better candidate for _future_ technical debt repayment.

File: core/commands/dht.go (hunk 1 above)
Comment (whyrusleeping) [inline core/commands/dht.go]: yeah, i opted for not making lots of changes to the codebase. we can do a codebase-wide change for that later

**Comment by whyrusleeping** [comment]
> bandaid on top of some already fairly hard to follow code

youre not wrong.
